### PR TITLE
Metapath additional axes

### PR DESCRIPTION
**Pull request by david-waltermire**
# Committer Notes

Added preceding, preceding-sibling, following, and following-sibling axis [support in Metapath](https://www.w3.org/TR/xpath-31/#axes).

Also replaced JMock with Mockito for better mocking support.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/metaschema-framework/metaschema-java/blob/main/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/metaschema-framework/metaschema-java/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all website and readme documentation affected by the changes you made?

### CODE REVIEW

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/INodeItem.java [hunk 1]
```diff
@@ -195,6 +210,48 @@ default Stream<? extends INodeItem> descendantOrSelf() {
...
+
+  /**
+   * Get the children of this node's parent, and their descendants, that occur
+   * before this node in a depth-first order.
+   *
+   * @return a stream of nodes
+   */
+  default Stream<? extends IModelNodeItem<?, ?>> following() {
+    return ObjectUtils.notNull(Stream.empty());
+  }
```
Comment (aj-stein-gsa) [inline core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/INodeItem.java:243]: Non-blocking: I am having trouble understanding how the functions for these different axes all return empty streams, am I misunderstanding that node items do not have preceding and follow siblings?

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/IModelNodeItem.java [hunk 2]
```diff
@@ -30,4 +25,46 @@ public interface IModelNodeItem<D extends IModelDefinition, I extends INamedMode
...
+  default Stream<? extends IModelNodeItem<?, ?>> following() {
+    return followingSibling()
+        .flatMap(IModelNodeItem::descendantOrSelf);
+  }
+
+  @Override
+  default Stream<? extends IModelNodeItem<?, ?>> preceding() {
+    return precedingSibling()
+        .flatMap(IModelNodeItem::descendantOrSelf);
+  }
```
Comment (aj-stein-gsa) [inline core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/IModelNodeItem.java:69]: Non-blocking: I am not understanding how this code works if the calling functions just do a null check but also stream a `Stream.empty()`, but it seems the tests

**Review by aj-stein-gsa — APPROVED** [review]

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/IModelNodeItem.java (hunk 2 above)
Comment (david-waltermire) [reply core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/IModelNodeItem.java:69]: the `precedingSibling` method implements the `preceding-sibling` access. The `descendantOrSelf` method is used to descend into the graph to pick up the rest of the items that are part of the `preceding` axis. `flatMap` replace an item in the stream, with items in the resulting stream generated by the lambda call.

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/INodeItem.java (hunk 1 above)
Comment (david-waltermire) [reply core/src/main/java/gov/nist/secauto/metaschema/core/metapath/item/node/INodeItem.java:243]: Most node types (i.e. flag, document) do not require returning anything when these methods are called. Onlu IModelNodeItem requires generating these nodes for the axes according to the [Metapath spec](https://www.w3.org/TR/xpath-31/#axes).

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**